pkg/scheduler/core/generic_scheduler.go

```diff
@@ -635,7 +635,9 @@ func (g *genericScheduler) podFitsOnNode(
 ) (bool, []predicates.PredicateFailureReason, *framework.Status, error) {
 	var failedPredicates []predicates.PredicateFailureReason
 	var status *framework.Status
-
+	if info.Node() == nil {
+		return false, []predicates.PredicateFailureReason{}, framework.NewStatus(framework.UnschedulableAndUnresolvable, "node was nil"), fmt.Errorf("unexpected nil node in nodeInfo")
```

Comment:
We shouldn't return an error, just UnschedulableAndUnresolvable status because and error will cause the whole scheduling cycle to fail, which we don't want to happen.